<?xml version="1.0"?>

<launch>
    <arg name="max_motor_speed"             default="256" />
    <arg name="pulses_per_meter"            default="536" />
    <arg name="wheel_separation_width"      default="0.58" />
    <arg name="wheel_separation_length"     default="0.65" />
    <arg name="rate"                        default="10.0" />
    
    <node pkg="rosserial_arduino"       type="serial_node.py"           name="serial_node"              output="screen">
        <param name="port"  type="string"   value="/dev/ttyACM0"/>
        <param name="baud"  type="int"      value="57600"/>
    </node>
    
    <node pkg="jiqi_mdc" type="mecanum_drive_controller" name="mecanum_drive_controller" output="screen">
        <param name="max_motor_speed"           type="int"      value="$(arg max_motor_speed)"/>
        <param name="pulses_per_meter"          type="int"      value="$(arg pulses_per_meter)"/>
        <param name="wheel_separation_width"    type="double"   value="$(arg wheel_separation_width)"/>
        <param name="wheel_separation_length"   type="double"   value="$(arg wheel_separation_length)"/>
        <param name="rate"                      type="double"   value="$(arg rate)"/>
    </node>

    <!-- Encoder Odometry -->
    <!-- Subscriptions: /encoder_pulse_data -->
    <!-- Publications:  [ENCODER DATA CONVERTED], [POTENTIALLY IMU & POINTCLOUD DATA IF CONVERSION NEEDED] -->
    <node pkg="jiqi_mdc" type="mecanum_drive_odometry" name="mecanum_drive_odometry" output="screen" >
        <rosparam command="load" file="$(find jiqi_navigation)/config/jiqi_mdc/mecanum_drive_odometry.yaml" />
    </node>
    s
    <include file="$(find articubot)/launch/pi_move_base.launch"></include>
    
</launch>
